File: Swift/reverse-integer.swift
class Solution {
    func reverse(_ x: Int) -> Int {
        var x = x
        var res = 0
        
        while x != 0 {
            res = res * 10 + x % 10
            x /= 10
        }
        
        return res > Int32.max || res < Int32.min ? 0 : res
    }
}

File: Python/reverse-integer.py
import sys

class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0:
            return -self.reverse(-x)

        res = 0
        while x:
            res = res * 10 + x % 10
            x //= 10
        return res if res <= 0x7fffffff else 0
